queries.sql

```diff
@@ -0,0 +1,94 @@
+-- Get all products of selected brand.
+CREATE OR REPLACE FUNCTION get_products_by_brand_id (search_id bigint) 
+RETURNS TABLE (
+  id bigint,
+  brand_id bigint,
+  category_id bigint,
+  name text,
+  price money,
+  quantity bigint,
+  created_at timestamp)
+  language plpgsql
+AS $$
+BEGIN
+  RETURN QUERY 
+    SELECT * FROM products WHERE products.brand_id = search_id;
+END;$$;
+
+-- Select all brands with the number of their products
+-- respectively. Order by the number of products.
+CREATE OR REPLACE FUNCTION get_brands_with_products_amount()
+RETURNS TABLE (
+  brand_id bigint,
+  amount numeric)
+  language plpgsql
+AS $$
+BEGIN
+  RETURN QUERY 
+    SELECT brands.id AS brand_id, SUM(quantity) AS amount
+    FROM brands JOIN products 
+    ON brands.id = products.brand_id
+    GROUP BY brands.id
+    ORDER BY amount DESC;
+END;$$;
+
+-- Get all products for a given category and section.
+CREATE OR REPLACE FUNCTION get_products_by_section_and_category(search_section_id bigint, search_category_id bigint)
+RETURNS TABLE (
+  id bigint,
+  brand_id bigint,
+  category_id bigint,
+  name text,
+  price money,
+  quantity bigint,
+  created_at timestamp)
+  language plpgsql
+AS $$
+BEGIN
+  RETURN QUERY 
+    SELECT products.id, products.brand_id, products.category_id, products.name, products.price, products.quantity, products.created_at
+    FROM products
+    INNER JOIN sections_categories ON products.category_id = sections_categories.category_id
+    GROUP BY products.id, sections_categories.section_id, sections_categories.category_id
+    HAVING sections_categories.section_id = search_section_id AND sections_categories.category_id = search_category_id;
```

Comment:
I think `group by` is redundant in this case. Instead of it you can put `section_id` filter to the `join` statement and also add `where` statement to filter out products by `category_id`